zetaclient/chains/evm/signer/signer.go

```diff
@@ -48,39 +47,57 @@ var (
 
 // Signer deals with the signing EVM transactions and implements the ChainSigner interface
 type Signer struct {
-	client      interfaces.EVMRPCClient
-	chain       *chains.Chain
-	tssSigner   interfaces.TSSSigner
-	ethSigner   ethtypes.Signer
-	logger      base.Logger
-	ts          *metrics.TelemetryServer
-	coreContext *clientcontext.ZetacoreContext
-
-	// mu protects below fields from concurrent access
-	mu                        *sync.Mutex
-	zetaConnectorABI          abi.ABI
-	erc20CustodyABI           abi.ABI
-	zetaConnectorAddress      ethcommon.Address
-	er20CustodyAddress        ethcommon.Address
+	// base.Signer implements the base chain signer
+	base.Signer
+
+	// client is the EVM RPC client to interact with the EVM chain
+	client interfaces.EVMRPCClient
+
+	// ethSigner encapsulates EVM transaction signature handling
+	ethSigner ethtypes.Signer
+
+	// zetaConnectorABI is the ABI of the ZetaConnector contract
+	zetaConnectorABI abi.ABI
+
+	// erc20CustodyABI is the ABI of the ERC20Custody contract
+	erc20CustodyABI abi.ABI
+
+	// zetaConnectorAddress is the address of the ZetaConnector contract
+	zetaConnectorAddress ethcommon.Address
+
+	// er20CustodyAddress is the address of the ERC20Custody contract
+	er20CustodyAddress ethcommon.Address
+
+	// outboundHashBeingReported is a map of outboundHash being reported
 	outboundHashBeingReported map[string]bool
+
+	// mu protects fields from concurrent access
+	mu *sync.Mutex
```

Comment:
Couldn't it be part of base signer as well?

Reply:
Having a mutex in base-level struct is completely possible. Keep in mind, only the sub-class (now the base class) knows when it needs to be used. Let's do it and add a description.

Reply:
https://github.com/zeta-chain/node/pull/2357/commits/e585bc0249a23d140a11e94c4be327d0075a6e8c